views/journal/talkform.tt

```diff
@@ -219,98 +219,83 @@ the same terms as Perl itself.  For a copy of the license, please reference
-          </td>
-          <td align='left'>
-            <b>
-              <label for='talkpostfromlj' onclick='handleRadios(2); return false;'>
-                [%- '/talkpost.bml.opt.siteuser' | ml( sitename => site.nameshort ) -%]
-              </label>
-            </b>
+        <div class="from-option" id="from-user-loggedout">
+
+            <input type='radio' name='usertype' value='user' data-more="from-user-more" id='talkpostfromlj'[% IF default_usertype == 'user' %] checked='checked'[% END %] />
+
+            <label for='talkpostfromlj'>
+              [% dw.img( 'id_user', '' ) %]
+              [% IF remote && ! remote.openid_identity %] Other [%- END %]
+              [% '/talkpost.bml.opt.siteuser' | ml( sitename => site.nameshort ) %]
+            </label>
             [%- IF journal.screens_all -%]
               [%- '/talkpost.bml.opt.willscreen' | ml -%]
             [%- ELSIF journal.screens_non_access -%]
               [%- '/talkpost.bml.opt.willscreenfriend' | ml -%]
             [%- END -%]
-          </td>
-        </tr>
+        </div>
 
         [%# site user login form, always present but sometimes hidden -%]
-        <tr valign='middle' align='left' id='lj_more'>
-          <td colspan='2'></td>
-
-          <td>
-            <table summary=''>
-              <tr>
-                <td>
-                  [%- 'Username' | ml -%][%# NF: ??? -%]
-                </td>
-                <td>
-                  [%- form.textbox(
-                    name = 'userpost'
-                    id = 'username'
-                    size = 13
-                    maxlength = 25
-                    value = ( comment.user || remote.user )
-                    onclick = 'this.value=""'
-                    style = "background: url('${site.imgroot}/silk/identity/user.png') no-repeat; background-color: #fff; background-position: 0px 1px; padding-left: 18px; color: #00C; font-weight: bold;"
-                  ) -%]
-                </td>
-              </tr>
-
-              <tr>
-                <td>
-                  [%- 'Password' | ml -%][%# NF: ??? -%]
-                </td>
-                <td>
-                  <input class='textbox' name='password' type='password' maxlength='30' size='18' id='password' />
-                </td>
-              </tr>
-
-              <tr>
-                <td colspan='2'>
-                  <label for='logincheck'>[% '/talkpost.bml.loginq' | ml %]&nbsp;</label>
-                  <input type='checkbox' name='do_login' id='logincheck' />
-                </td>
-              </tr>
-            </table>
-          </td>
-        </tr>
+        <div class="from-login" id="from-user-more">
+
+          <div>
+            [%- form.textbox(
+              name = 'userpost'
+              id = 'username'
+              label = dw.ml('Username')
+              size = 13
+              maxlength = 25
```

Comment:
We recently landed a diff to pull these "magic constants" out and use the $LJ::MAX_USERNAME_LENGTH or whatever the constant is. Would be nice to do that, might need to add it to the LJ::Web thingy too?

Reply:
#2456 is the PR being referenced.

Reply:
oh hey good call

Reply:
Can we put that here? https://github.com/dreamwidth/dw-free/blob/master/cgi-bin/DW/Template.pm#L37 Or should it be wired through the template arguments?

Reply:
- wired through template is the practice in the previous PR. 
- lj::web thingy -- quickreply doesn't need a username, and idk if there's any javascript that has a use for it.